test/test_fat.rb

```diff
@@ -5,6 +5,40 @@
 class FatFileTest < Minitest::Test
   include Helpers
 
+  def test_empty_file
+    empty_file = tempfile_with_data("empty_file", "")
+
+    assert_raises MachO::NotAMachOError do
```

Comment:
Maybe check for the more specific exception classes like `TruncatedFileError` here and in the other newly added test cases?